### PR TITLE
Initializing and configuring project

### PR DESCRIPTION
**Pull request by aleklisi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - mix compile
+    - MIX_ENV=test mix compile
+
+env:
+    - PRESET=test
+
+elixir:
+    - '1.6.6'
+otp_release:
+    - '20.3.6'
```
Comment (rslota) [inline .travis.yml]: Please add a new line at the end of file - https://stackoverflow.com/a/729795/5663848 :)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+language: elixir
+
+sudo: false
+
+install:
+    - mix local.rebar --force
+    - mix local.hex --force
+    - mix deps.get
+    - mix compile
```
Comment (rslota) [inline .travis.yml]: I'm pretty sure we don't need to compile twice. We only need a `test` build.

File: .travis.yml (hunk 1 above)
Comment (rslota) [inline .travis.yml]: We should also (or only) add `OTP-21.0` unless Travis doesn't support it yet.

File: .travis/script.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        MIX_ENV=test mix coveralls.travis --include system
+        ;;
+    dialyzer)
+        mix dialyzer
+        ;;
+    *)
+        echo "Invalid preset: $PRESET"
+        exit 1
+        ;;
+esac
```
Comment (rslota) [inline .travis/script.sh]: Please add a new line at the end of file - https://stackoverflow.com/a/729795/5663848 :)

File: .travis.yml (hunk 2 above)
Comment (aleklisi) [reply .travis.yml]: copied from https://github.com/esl/MongooseICE/blob/4c2792b2e01ccff377c8f78b131a5ca2b064f39b/.travis.yml#L12

Comment (rslota) [reply .travis.yml]: Then let's fix it here at least :)

**Pull request merged**